### ISSUE TITLE
Write an example RL application using Stable Baseline3

### ISSUE THREAD

**Issue by XyDrKRulof** [upb-lea/gym-electric-motor#70]
Write an example application with the deep Q-network (DQN) implementation of Stable Baselines 3. Solve the current control problem of the permanent magnet synchronous motors (PMSM).

- https://github.com/DLR-RM/stable-baselines3
- Use mean absolute error (MAE) reward function
- Single discrete action variable: fundamental voltage vector with 7 options
- Cont. state variables

Take notes on the usability of the library and other notable findings. Prepare a small presentation of your results.

**Comment by XyDrKRulof** [comment]
In the branch "issue_#70" is now a working version of the stable baselines3 DQN with our PMSM. Here a small view of the results:

![expected](https://user-images.githubusercontent.com/65008699/88781595-e1138180-d18c-11ea-8c74-1d122e2d29ad.png)

For the routine I did following adjustments:

- Used the hotfix provided for issue #76 in its corresponding branch
- Limit an episode length to 10000 timesteps (0.1s) to prevent overfitting on motor speeds (gym TimeLimit wrapper)
- Flattened the observation/reference Tuple to fit into stable baselines3 (gym FlattenObservation wrapper)
- Wrote an own wrapper that changes epsilon to cos(epsilon) and sin(epsilon)
- Decreased env.action_space from 8 to 7 since action 0 and 7 are equal in effect

For the hyperparameters I used the following:
- Training steps = 500000 (5s)
- Memory warmup = 10000 steps
- Memory size = 200000 steps
- Gamma = 0.99
- Learning_rate = 1e-4
- Optimizer = Adam
- Frequency of mini-batch sampling and updating: each step
- Mini-Batch-size: 25
- Network architecture: Fully connected [17, 64, 64, 7] where 17 is the flattened observation+reference and 7 is the number of actions
- Network activation functions: ReLU
- Target model update = Every 1000 steps
- Epsilon-decay: From 1 to 0.05 in the first 10% of the training steps (so it's 0.05 past 50k steps)

**Comment by XyDrKRulof** [comment]
Small remark: If you want to execute the example yourself you shouldn't install stable baselines3 using pip as that version is outdated. Install it directly from their git repository (https://github.com/DLR-RM/stable-baselines3)

**Comment by XyDrKRulof** [comment]
Added random state initalization at the beginning of each episode:

![randomstateinit](https://user-images.githubusercontent.com/65008699/90608962-a341d000-e203-11ea-90cf-b45e25f0763d.png)

Also measured execution time on neutral machine: 1148.81s (19m 8s)
